src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -35,16 +35,14 @@ public static IObjectGraphType GetOperationRootType(Document document, ISchema s
                     type = schema.Subscription;
                     if (type == null)
                     {
-                        error = new ExecutionError("Schema is not configured for subscriptions");
+                        error = new InvalidOperationError("Schema is not configured for subscriptions");
                         error.AddLocation(operation, document);
                         throw error;
                     }
                     break;
 
                 default:
-                    error = new ExecutionError("Can only execute queries, mutations and subscriptions.");
-                    error.AddLocation(operation, document);
-                    throw error;
+                    throw new ArgumentOutOfRangeException(nameof(operation), "Can only execute queries, mutations and subscriptions.");
```

Comment:
That is, we consider this situation extremely non-standard and now we throw an exception instead of `ExecutionError` with filled location?

Reply:
The enum only contains the three members, and this is fully checked by the parser. If somehow this was called with another enum then it's an invalid argument to the function.

Reply:
OK